library/dnet.c

```diff
@@ -433,12 +433,13 @@ static int dnet_iterator_server_send_complete(struct dnet_addr *addr, struct dne
 		err = cmd->status;
 
 		/*
-		 * Write CAS error is not a real 'error' in that regard, that we do not remove
-		 * local record, but also do not stop iterator.
-		 *
 		 * Setting @write_error forces iterator to stop.
+		 *
+		 * Stop iteration only if connection was failed on write
+		 * or no space left on remote backend.
 		 */
-		if (err && !send->write_error && (err != -EBADFD))
+		const int critical_error = err && (err == -ENXIO || err == -ENOSPC);
```

Comment:
I believe, timeout has to stop iterator - receiving side does not cope with the data rate.
I do not know when write can return ENOENT